src/ml_gym/gym/trainer.py

```diff
@@ -48,19 +48,22 @@ def check_if_initiable_component(loss_fun: Loss):
             self.logger.log(LogLevel.INFO, "Running warmup...")
             with torch.no_grad():
                 prediction_batches = self.map_batches(fun=self.forward_batch,
-                                                      fun_params={"device": device, "model": model},
+                                                      fun_params={"calculation_device": device, "model": model,
+                                                                  "result_device": torch.device("cpu")},
```

Comment:
Should the result_device be hardcoded? We could also take it from the config or?

Reply:
When we create the InferenceResultBatch, the tensors don't have to be on the GPU anymore, because the training step is done. It's just used to calculates metrics etc. from that point on.